comp/core/secrets/component.go

```diff
@@ -0,0 +1,25 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package secrets decodes secret values by invoking the configured executable command
+package secrets
+
+import (
+	"io"
+)
+
+// team: agent-shared-components
+
+// Component is the component type.
+type Component interface {
+	// Configure the executable command that is used for decoding secrets
+	Configure(command string, arguments []string, timeout, maxSize int, groupExecPerm, removeLinebreak bool)
+	// Get debug information and write it to the parameter
+	GetDebugInfo(w io.Writer)
+	// Whether this component is enabled, if disabled other methods will only log and error and return
+	IsEnabled() bool
```

Comment:
Ideally a component that can be enabled or not should be implemented using `Optional[Component]`. See [this](https://github.com/DataDog/datadog-agent/blob/470e8dfc63c4d27f310c989ad0ee2bebf9afa407/comp/logs/agent/agent.go#L85-L111) as an example. Would it be relevant in the case of secrets?

Reply:
`secrets` is not a product, like `log`, but a building block for other pieces of code. So instead of checking N time across all the code base if `secrets` was enabled by the main I think it's better to have `secrets` always available but resolve secrets only when enabled.

That being said, other comp/pkg should not have to wonder if secrets is enabled or not. I could not find any call to this method so  we could remove this method (which indeed reproduces the `Optional` type we already have).

Reply:
The call was in `setupConfig` in comp/core/config/setup.go, which decides whether to call `config.LoadWithoutSecret` or `config.LoadWithSecret` based upon the `enabled` flag. I agree that the code is better without it, just having a disabled secretResolver means that no decrypted ever happen, without call-sites needing to know about it.